app/controllers/admin/project_logs_controller.rb

```diff
@@ -1,12 +1,51 @@
 class Admin::ProjectLogsController < Admin::AdminController
-  def destroy
+  include TranslationSaveable
+
+  def show
     @log = ProjectLog.find(params[:id])
-    @step = ProjectStep.find(@log.project_step_id)
-    @loan = Loan.find(@step.project_id)
+    authorize_with_parents
 
-    authorize @log
-    authorize @step
-    authorize @loan
+    redirect_to admin_loan_path(@loan)
+  end
+
+  def new
+    @log = ProjectLog.new(project_step_id: params[:step_id])
+    authorize_with_parents
+    @progress_metrics = ProjectLog.progress_metric_options
+    render "admin/logs/new", layout: false
+  end
+
+  def edit
+    @log = ProjectLog.find(params[:id])
+    authorize_with_parents
+    @progress_metrics = ProjectLog.progress_metric_options
+    render "admin/logs/edit", layout: false, locals: {log: @log}
```

Comment:
You don't need to pass `locals: `. Note that the `log` local is not actually used anywhere.

Remember the rule:

Only pass a local if the template is going to be used in multiple places on the page, which means you can't use an instance variable (`@foo`). Prefer instance variables where possible.